components/camel-aws/src/main/java/org/apache/camel/component/aws/common/AwsExchangeUtil.java

```diff
@@ -26,9 +26,7 @@ private AwsExchangeUtil() {
 
     public static Message getMessageForResponse(final Exchange exchange) {
         if (exchange.getPattern().isOutCapable()) {
-            Message out = exchange.getOut();
-            out.copyFrom(exchange.getIn());
-            return out;
+            return exchange.getOut();
         }
         return exchange.getIn();
     }
```

Comment:
Ok, I apologize for not being thorough with the testing.  I thought a
simple fix may fix this problem but I was finally able to fix the issue by
not changing the lower level code but rather directly address the multiple
calls to the addToResult call.  I have run the unit test and it passes
now.  Plus I have verified the code does properly work now.  I will apply
for another pull request.

thanks

won

On Wed, Apr 27, 2016 at 4:14 AM, Claus Ibsen <notifications@github.com>
wrote:

> In
> components/camel-aws/src/main/java/org/apache/camel/component/aws/common/AwsExchangeUtil.java
> <https://github.com/apache/camel/pull/961#discussion_r61240594>:
>
> > @@ -26,9 +26,7 @@ private AwsExchangeUtil() {
> >
> >      public static Message getMessageForResponse(final Exchange exchange) {
> >          if (exchange.getPattern().isOutCapable()) {
> > -            Message out = exchange.getOut();
> > -            out.copyFrom(exchange.getIn());
> > -            return out;
> > +            return exchange.getOut();
> >          }
> >          return exchange.getIn();
> >      }
>
> You should do
>
> if (exchange.hasOut()) { return exchange.getOut(); } else { return exchange.getIn(); }
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/camel/pull/961/files/f9c65d0c621206a316ea81a9b4e3164d009dbe1c#r61240594>
>



-- 
*(at)urbanenomad.com